pyaedt/modeler/cad/Modeler.py

```diff
@@ -1134,6 +1207,388 @@ def _attributes(self):
+        entity : int, VertexPrimitive, EdgePrimitive, FacePrimitive
+            Entity where the Object Coordinate System is anchored.
+            It can be the Id of the object or an object of type VertexPrimitive, EdgePrimitive, FacePrimitive.
+        x_axis_entity : int, VertexPrimitive, EdgePrimitive, FacePrimitive
+            Entity where the x axis of the Object Coordinate system points to.
+            It can be the Id of the object or an object of type VertexPrimitive, EdgePrimitive, FacePrimitive.
+            x_axis_entity has to be provided only if cs_type is set to ``Both``.
+        move_to_end : bool, optional
+            If ``True`` the Coordinate System creation operation will always be moved to the end of subsequent
+            objects operation. This will guarantee that the coordinate system will remain solidal with the object
+            face. If ``False`` the option "Always Move CS to End" is set to off. The default is ``True``.
+        reverse_x_axis : bool, optional
+            Whether the x-axis is in the reverse direction.
+            The default is ``False``.
+        reverse_y_axis : bool, optional
+            Whether the y-axis is in the reverse direction.
+            The default is ``False``.
+        # y_axis_x_direction : float, optional
+        #     X direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+        # y_axis_y_direction : float, optional
+        #     Y direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+        # y_axis_z_direction : float, optional
+        #     Z direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        """
+        if cs_type == "Offset" or cs_type == "Both":
+            is_attached_to_entity = True
+            origin_entity_id = self._modeler.convert_to_selections(entity, True)[0]
+            if not isinstance(origin_entity_id, int):  # pragma: no cover
+                raise ValueError("Unable to find reference entity.")
+            else:
+                self.entity_id = origin_entity_id
+                o_type = self._get_type_from_id(self.entity_id)
+                if o_type == "Face":
+                    origin_position_type = "FaceCenter"
+                elif o_type == "Edge":
+                    origin_position_type = "EdgeCenter"
+                elif o_type == "Vertex":
+                    origin_position_type = "OnVertex"
+                else:  # pragma: no cover
+                    raise ValueError("origin must identify either Face or Edge or Vertex.")
```

Comment:
```suggestion
                    raise ValueError("Origin must identify either `Face`, 'Edge', or 'Vertex'.")
```
Message strings should be complete sentences. If you don't want to tag the options as possible values, use lowercase letters.